### PR TITLE
ELPP-1889 Added article interface methods (and implemented them)

### PR DESCRIPTION
**Pull request by stephenwf**
@thewilkybarkid  @giorgiosironi

### CODE REVIEW

File: src/Model/ExternalArticle.php [hunk 1]
```diff
@@ -39,4 +39,19 @@ public function getUri(): string
...
+    }
+
+    public function getType(): string
+    {
+        return 'external-article';
+    }
+
+    public function getTitle(): string
+    {
+        return $this->getArticleTitle();
```
Comment (thewilkybarkid) [inline src/Model/ExternalArticle.php]: I'd just rename `getArticleTitle()` (the object is called `ExternalArticle`, so it's clearly the title of the article).

File: src/Model/Article.php [hunk 2]
```diff
@@ -4,4 +4,9 @@
 
 interface Article extends Model
```
Comment (thewilkybarkid) [inline src/Model/Article.php]: Implement `HasId` (so `getId()` overrides it to define the return type).

**Comment by stephenwf** [comment]
@thewilkybarkid added both of those 👍

**Comment by thewilkybarkid** [comment]
`getId()` and `getType()` will need tests.

I think we should match the normal [eLife ID format](https://github.com/elifesciences/api-raml/blob/develop/src/misc/id.v1.yaml#L4) rather than just using the plain URI too (in case we start using this elsewhere). Maybe `external-[URI hash]`?

**Comment by stephenwf** [comment]
@thewilkybarkid okay so the ID is now `"external-" . sha1(uri)`

**Comment by giorgiosironi** [comment]
Better to rebase on `master`.

**Comment by stephenwf** [comment]
Rebased now

**Pull request merged**